Description: Call taglib_file_is_valid() for input audio files
 Check that the file contains valid audio information to determine whether
 the plugin should act on it or not.
Origin: other, http://bugzilla.xfce.org/attachment.cgi?id=2879
Bug: http://bugzilla.xfce.org/show_bug.cgi?id=6297
---
 thunar-plugin/media-tags-provider.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/thunar-plugin/media-tags-provider.c b/thunar-plugin/media-tags-provider.c
index a7604ea..2c31a98 100644
--- a/thunar-plugin/media-tags-provider.c
+++ b/thunar-plugin/media-tags-provider.c
@@ -163,10 +163,11 @@ media_tags_get_audio_file_supported (ThunarxFileInfo *info)
 
   if (G_LIKELY (taglib_file != NULL))
     {
+      /* Check that the file contains valid audio info */
+      supported = taglib_file_is_valid (taglib_file);
+
       /* Free the taglib file */
       taglib_file_free (taglib_file);
-
-      supported = TRUE;
     }
 
   g_free (filename);

